### ISSUE TITLE
Check cookie usage

### ISSUE THREAD

**Issue by emmajclegg** [younginnovations/iatipublisher#1456]
Following discussion in [#1450](https://github.com/younginnovations/iatipublisher/issues/1450), we believe IATI Publisher should have a banner message to inform users of cookie usage. It is good practice to declare essential cookies and we may need to provide an opt-out for IP address and browser information collection.

(from YI):
- Session Cookies: These are used for managing user sessions, allowing users to remain logged in.
- CSRF Token Cookies: These are utilized for security purposes, protecting against Cross-Site Request Forgery (CSRF) attacks to ensure that all actions performed on the site are authorized.

Besides the mentioned cookies we also collect standard browsing data (IP Address and Browser Information) for security and analytical purposes.

--------

Tagging @robredpath to edit or provide more info on this (thanks)

**Comment by robredpath** [comment]
In the UK there are two pieces of legislation that are relevant here - PECR ("cookie law") and GDPR ("privacy law"). 

Information about PECR can be found [on the ICO (the UK's privacy regulator) website here](https://ico.org.uk/for-organisations/direct-marketing-and-privacy-and-electronic-communications/guide-to-pecr/guidance-on-the-use-of-cookies-and-similar-technologies/what-are-the-rules-on-cookies-and-similar-technologies/). For IATI Publisher, the cookies described are clearly essential to the operation of the website, so we don't need to seek consent. It's good practice to have a cookie policy that describes the cookies that are used. IATI's [cookie policy](https://iatistandard.org/en/privacy-policy/cookie-policy/) is linked from our [privacy policy](https://iatistandard.org/en/privacy-policy/). 

Tracking individual users (i.e. to the level of being able to say that a particular user clicked on this, then went to that page, then did this, then returned a week later, and so on) is legally complex as it brings in aspects of PECR and GDPR. We don't need that level of user tracking for our purposes, so we don't do it. We have [Plausible](https://plausible.io/publisher.iatistandard.org/) set up for analytics; I don't think we need anything else. Logging per-request information (such as browser user-agent strings and IP addresses) is fine as long as we don't use it to try and track an individual user, and we don't retain it any longer than we need for our purposes. 

So, tl;dr: link to [IATI's privacy policy](https://iatistandard.org/en/privacy-policy/) on every page in IATI Publisher, don't track users, and I think we're fine. 

(and yes, IATI's privacy policy needs a lot of work, but it will do for now)

**Comment by emmajclegg** [comment]
Thanks @robredpath

@praweshsth - it sounds like IATI Publisher doesn't need additional messaging about cookies added. Can YI please check the level of user tracking based on Rob's message above, and ensure that the [privacy policy](https://iatistandard.org/en/privacy-policy/) link is included in the footer of every IATI Publisher page? (it doesn't look like it is currently)

**Comment by PG-Momik** [comment]
@emmajclegg 
> Tracking individual users (i.e. to the level of being able to say that a particular user clicked on this, then went to that page, then did this, then returned a week later, and so on) is legally complex as it brings in aspects of PECR and GDPR.

While we do not track mouse clicks and movements, we do log when users perform certain actions such as save, update, delete, login,download, and export. These logs cannot be used for any kind of user profiling, nor be able to say that a particular user clicked where.

IP addresses and browser client information can be considered "personally identifiable" in some contexts, but this is debatable. If preferred, we can stop logging IP addresses and browser client information. Additionally we can remove logs older than 1 month.

These measures can help us enhance our GDPR compliance by minimizing data collection and retention.

**Comment by emmajclegg** [comment]
Thanks for the extra info @PG-Momik. If you currently use IP address/browser information for something other than tracking individual users, I think that's fine based on Rob's description above. Deleting logs after 1 month sounds reasonable. If you're not using the information for anything, then it's probably best to stop logging it.

For including the privacy policy link on all IATI Publisher pages, feel free to reopen the previous issue [#1450](https://github.com/younginnovations/iatipublisher/issues/1450) or include it in this current one, whatever's easiest. I'm not sure what end users can see but I can't see the privacy policy in the page footer when proxied into an organisation's account for example.

**Comment by PG-Momik** [comment]
@emmajclegg 
This has been deployed to staging. Privacy policy link is now available in all the pages. User logs will only be stored for 1 month.
